### ISSUE TITLE
Compatible release clause will bypass downgrade_blacklist

### ISSUE THREAD

**Issue by sko00o** [Comfy-Org/ComfyUI-Manager#1301]
I noticed some custom_nodes using `~=` in requirements.txt .

It will make downgrade_blacklist not work.

## How to reproduce this

1. set diffusers in downgrade_blacklist

```ini
# ComfyUI/custom_nodes/ComfyUI-Manager/config.ini
[default]
# ...
downgrade_blacklist = diffusers
```

2. install diffusers in venv

```
$ pip install diffusers==0.27.2
$ pip list | grep diffusers
diffusers                         0.27.2
```

3. restart ComfyUI

4. Install custom_nodes [StableZero123-comfyui](https://github.com/deroberon/StableZero123-comfyui)

The requirements.txt in this repo has this [line](https://github.com/deroberon/StableZero123-comfyui/blob/main/requirements.txt#L1) `diffusers~=0.21.4`

5. check diffusers version again

```
$ pip list | grep diffusers
diffusers                         0.21.4
```

**Comment by ltdrdata** [comment]
I cannot reproduce that issue.
Is your ComfyUI up to date?

**Comment by sko00o** [comment]
> Is your ComfyUI up to date?

Yes, I'm using the latest ComfyUI. And the issue is not related to ComfyUI.

---

Sorry for missing one step in bug reproduction. I should install a higher version diffusers in Step 2. 
Considering the ComfyUI does not rely on diffusers. I simply use `pip install` to mock some custom nodes installation behavior.

For demonstration, I built a docker image with latest ComfyUI and ComfyUI-Manager installed. (https://github.com/sko00o/playground/tree/main/comfyui)

Open this repo in [Github Codespace](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=558393073), and follow the steps:

```sh
# 1. launch comfyui and downgrade_blacklist will be set automatic
sudo bash comfyui/run.sh

# 2. install diffusers
pip install diffusers==0.27.2
pip list | grep diffusers
# diffusers                 0.27.2

# 3. open WebUI to install `StableZero123-comfyui` by ComfyUI-Manager

# 4. check diffusers version again
pip list | grep diffusers
# diffusers                 0.21.4
```

**Comment by sko00o** [comment]
PTAL  @ltdrdata

**Comment by ltdrdata** [comment]
confirmed the issue.

**Comment by ltdrdata** [comment]
This is fixed.
